### PR TITLE
Add support for log scaling of maps, as well as input normalizations.

### PR DESCRIPTION
**Pull request by erykoff**

### CODE REVIEW

**Comment by erykoff** [comment]
This PR additionally removes support for now very old matplotlib.

File: skyproj/_docstrings.py [hunk 1]
```diff
@@ -80,6 +80,10 @@ def skyproj_init_parameters(intro_string, include_projection_name=False, include
     Longitude range to plot [``lon_min``, ``lon_max``].
 lat_range : `tuple` [`float`, `float`], optional
     Latitude range to plot [``lat_min``, ``lat_max``].
+norm : `str` or `matplotlib.colors.Normalize`, optional
+    The normalization method used to scale the data. By default a
+    linear scaling is used. This may be an instance of `Normalize` or
+    a scale name, such as ``linear``, ``log``, ``symlog``, ``logit``.
```
Comment (kadrlica) [inline skyproj/_docstrings.py:86]: 👍

File: skyproj/_skyproj.py [hunk 2]
```diff
@@ -1091,7 +1114,8 @@ def draw_polygon_file(self, filename, reverse=True,
 
     @add_func_docstr(draw_hpxmap_docstr)
     def draw_hpxmap(self, hpxmap, nest=False, zoom=True, xsize=1000, vmin=None, vmax=None,
-                    rasterized=True, lon_range=None, lat_range=None, **kwargs):
+                    rasterized=True, lon_range=None, lat_range=None,
+                    log_scale=False, norm="linear", **kwargs):
```
Comment (kadrlica) [inline skyproj/_skyproj.py:1118]: Do we need to track "log_scale" explicitly or can we determine it from the norm? What happens if the user provides norm="linear" and log_scale=True?

File: skyproj/_skyproj.py [hunk 3]
```diff
@@ -1156,7 +1190,8 @@ def draw_hpxmap(self, hpxmap, nest=False, zoom=True, xsize=1000, vmin=None, vmax
     @add_func_docstr(draw_hpxpix_docstr)
     def draw_hpxpix(self, nside, pixels, values, nest=False, zoom=True, xsize=1000,
                     vmin=None, vmax=None,
-                    rasterized=True, lon_range=None, lat_range=None, **kwargs):
+                    rasterized=True, lon_range=None, lat_range=None,
+                    log_scale=False, norm="linear", **kwargs):
```
Comment (kadrlica) [inline skyproj/_skyproj.py:1194]: Is "log_scale" needed?

**Review by kadrlica — COMMENTED** [review]
I'm not 100% sure, but I think that the "log_scale" argument in draw_hpxmap is no longer necessary. I don't think this argument exists for draw_hspmap.

File: skyproj/_skyproj.py [hunk 4]
```diff
@@ -1208,7 +1251,7 @@ def draw_hpxpix(self, nside, pixels, values, nest=False, zoom=True, xsize=1000,
     @add_func_docstr(draw_hspmap_docstr)
     def draw_hspmap(self, hspmap, zoom=True, xsize=1000, vmin=None, vmax=None,
                     rasterized=True, lon_range=None, lat_range=None, valid_mask=False,
-                    **kwargs):
+                    norm="linear", **kwargs):
```
Comment (kadrlica) [inline skyproj/_skyproj.py:1254]: There is no "log_scale=True" here

Comment (erykoff) [reply skyproj/_skyproj.py:1253]: Fixed!

File: skyproj/_skyproj.py (hunk 3 above)
Comment (erykoff) [reply skyproj/_skyproj.py]: Fixed!

File: skyproj/_skyproj.py (hunk 2 above)
Comment (erykoff) [reply skyproj/_skyproj.py]: Fixed!

**Review by kadrlica — APPROVED** [review]

**Pull request merged**